### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by offsset**
https://offsset.github.io/react_static-list-of-posts/

### CODE REVIEW

File: src/components/CommentInfo/CommentInfo.tsx [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+  name: string;
+  email: string;
+  body: string;
+};
+
+export const CommentInfo: React.FC<Props> = ({ name, email, body }) => {
+  const mail = `mailto:${email}`;
+
+  return (
+    <>
```
Comment (vlad-demchuk) [inline src/components/CommentInfo/CommentInfo.tsx:13]: you don't need this fragment

File: src/components/CommentList/CommentList.tsx [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+export const CommentList: React.FC<Props> = ({ comments }) => (
+  <>
+    <div className="CommentList">
+      <CommentInfo
+        name={comments.name}
+        email={comments.email}
+        body={comments.body}
+      />
+    </div>
+  </>
```
Comment (vlad-demchuk) [inline src/components/CommentList/CommentList.tsx:19]:
```suggestion
    <div className="CommentList">
      <CommentInfo
        name={comments.name}
        email={comments.email}
        body={comments.body}
      />
    </div>
```

File: src/components/PostList/PostList.tsx [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+
+      { post.map(content => (
+        content.user
+        && (
+          <PostInfo
+            title={content.title}
+            body={content.body}
+            name={content.user.name}
+            email={content.user.email}
+            comments={content.comments}
```
Comment (vlad-demchuk) [inline src/components/PostList/PostList.tsx:23]: use destructuring

**Review by vlad-demchuk — CHANGES_REQUESTED** [review]
fix linter issues (just push your types to repo)
image
image
image

remove redundant react fragments

File: src/components/PostInfo/PostInfo.tsx [hunk 4]
```diff
@@ -1,5 +1,58 @@
...
+import { UserInfo } from '../UserInfo';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
+type Props = {
+  title: string;
+  body: string;
+  user: User | null;
+  comments: Comments [];
+};
```
Comment (TarasHoliuk) [inline src/components/PostInfo/PostInfo.tsx:13]:
```suggestion
type Props = {
  post: {
    title: string;
    body: string;
    user: User | null;
    comments: Comments [];
  }
};
```

but better in this way:
```
type Props = {
  post: Post;
}
```

File: src/components/PostList/PostList.tsx [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+  post: Todo[];
+};
+
+export const PostList: React.FC<Props> = ({ post }) => (
+  <div className="PostList">
+
+    { post.map(content => (
+      content.user
+      && (
+        <PostInfo {...content} />
```
Comment (TarasHoliuk) [inline src/components/PostList/PostList.tsx:17]: don't pass props in this way
you should define one object in props and pass entire object

File: src/components/PostList/PostList.tsx [hunk 6]
```diff
@@ -1,5 +1,21 @@
...
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  post: Todo[];
+};
+
+export const PostList: React.FC<Props> = ({ post }) => (
+  <div className="PostList">
+
+    { post.map(content => (
```
Comment (TarasHoliuk) [inline src/components/PostList/PostList.tsx:14]:
```suggestion
    {post.map(content => (
```

File: src/components/PostInfo/PostInfo.tsx [hunk 7]
```diff
@@ -1,5 +1,58 @@
 import React from 'react';
+import './PostInfo.scss';
+import { Comments } from '../../types/Comments';
+import { User } from '../../types/User';
```
Comment (TarasHoliuk) [inline src/components/PostInfo/PostInfo.tsx:4]: you still didn't add it
type in terminal `git add .`, then commit and push
`git add .` will add newly created files

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]

**Review by TarasHoliuk — APPROVED** [review]

**Pull request closed without merging**